### PR TITLE
Set up support section in docs

### PR DESCRIPTION
**Pull request by energyLS**
## Changes proposed in this Pull Request
This PR revises the "contributions" section and adds the "support" section to include the link to stackoverflow and clarify the outreach channels.
Apart from the docs, it adds stackoverflow to the github issue template and readme.

## Checklist

- [x] Code changes are sufficiently documented; i.e. new functions contain docstrings and further explanations may be given in `doc`.
- [x] Unit tests for new features were added (if applicable).
- [x] Newly introduced dependencies are added to `environment.yaml`, `environment_docs.yaml` and `setup.py` (if applicable).
- [x] A note for the release notes `doc/release_notes.rst` of the upcoming release is included.
- [x] I consent to the release of this PR's code under the MIT license.

### CODE REVIEW

**Comment by energyLS** [comment]
@euronion I'd be happy to receive your review, can't add you though due to limited rights.

**Comment by energyLS** [comment]
@euronion I have chatted with @fneum who is in general fine with the proposed structure (didn't look into detail though). We have now added [shields](https://img.shields.io/stackexchange/stackoverflow/t/pypsa) to readme and docs.

File: README.md [hunk 1]
```diff
@@ -173,21 +174,14 @@ Please check the [documentation](https://pypsa.readthedocs.io).
...
-page](https://github.com/PyPSA/PyPSA/issues).
-
-A `Discord server <https://discord.gg/AnuJBk23FU>` hosts every tool
-in the PyPSA ecosystem. We have there public voice and text channels
-that are suitable to organise projects, ask questions,
-share news, or chat with the community.
+-   In case of code-related **questions**, please post on [stack overflow](https://stackoverflow.com/questions/tagged/pypsa).
+-   For non-programming related and more general questions please refer to the [mailing list](https://groups.google.com/group/pypsa).
+-   To **discuss** with other PyPSA users, organise projects, ask questions, share news, or chat with the community
+use the [discord server](https://discord.gg/JTdvaEBb).
```
Comment (euronion) [inline README.md:182]:
```suggestion
-   To **discuss** with other PyPSA users, organise projects, share news, and get in touch with the community you can use the [discord server](https://discord.gg/JTdvaEBb).
```

File: doc/support.rst [hunk 2]
```diff
@@ -0,0 +1,11 @@
+
+-------
+Support
+-------
+
+* In case of code-related **questions**, please post on `stack overflow <https://stackoverflow.com/questions/tagged/pypsa>`_.
+* For non-programming related and more general questions please refer to the `mailing list <https://groups.google.com/group/pypsa>`_.
+* To **discuss** with other PyPSA users, organise projects, ask questions, share news, or chat with the community
+use the `discord server <https://discord.gg/JTdvaEBb>`_.
```
Comment (euronion) [inline doc/support.rst:9]:
```suggestion
* To **discuss** with other PyPSA users, organise projects, share news, and get in touch with the community you can use the [discord server](https://discord.gg/JTdvaEBb).
```

File: doc/support.rst [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+* For **bugs and feature requests**, please use the `issue tracker <https://github.com/PyPSA/PyPSA/issues>`_.
+* For **contributions**, visit PyPSA on `Github <https://github.com/PyPSA/PyPSA>`_.
```
Comment (euronion) [inline doc/support.rst:11]:
```suggestion
* We strongly welcome anyone interested in providing **contributions** to this project. If you have any ideas, suggestions or encounter problems, feel invited to file issues or make pull requests on `Github <https://github.com/PyPSA/PyPSA>`_.
```

Consistency with README.md

**Review by euronion — CHANGES_REQUESTED** [review]
* Looking mostly good, thank you!

* Shields are always nice ;-)

* I think we can get rid of the "Contributing" section in the documentation then?

**Review by fneum — APPROVED** [review]
Thanks for taking care of this!

**Pull request merged**